run_demo.sh

```diff
@@ -40,33 +42,96 @@ function check_hostname(){
   fi
 }
 
-if [ -z "${1:-}" ]; then
-  echo "Usage: $0 <diracx_src_dir> [other source directories]"
-  exit 1
-fi
-diracx_repo_dir="$(readlink -f "${1}")"
-diracx_src_dir="${diracx_repo_dir}/src/diracx"
-if [[ ! -d "${diracx_src_dir}" ]]; then
-  printf "\U26A0\UFE0F Error: %s is not a clone of DiracX!" "${diracx_repo_dir}"
-  exit 1
-fi
+usage="${0##*/} [-h|--help] [--exit-when-done] [--] <diracx_src_dir> [other source directories]"
+
+# Parse command-line switches
+exit_when_done=0
+while [ -n "${1:-}" ]; do case $1 in
+	# Print a brief usage summary and exit
+	-h|--help|-\?)
+		printf 'Usage: %s\n' "$usage"
+		exit ;;
+
+	# # Unbundle short options
+	# -[niladic-short-opts]?*)
+	# 	tail="${1#??}"
+	# 	head=${1%"$tail"}
+	# 	shift
+	# 	set -- "$head" "-$tail" "$@"
+	# 	continue ;;
+
+	# # Expand parametric values
+	# -[monadic-short-opts]?*|--[!=]*=*)
+	# 	case $1 in
+	# 		--*) tail=${1#*=}; head=${1%%=*} ;;
+	# 		*)   tail=${1#??}; head=${1%"$tail"} ;;
+	# 	esac
+	# 	shift
+	# 	set -- "$head" "$tail" "$@"
+	# 	continue ;;
 
-declare -a pkg_dirs
-declare -a pkg_names
+	# Add new switch checks here
+	--exit-when-done)
+    exit_when_done=1;
+		shift
+		break ;;
+
+	# Double-dash: Terminate option parsing
+	--)
+		shift
+		break ;;
+
+	# Invalid option: abort
+	--*|-?*)
+		>&2 printf '%b %s: Invalid option: "%s"\n' ${SKULL_EMOJI} "${0##*/}" "$1"
+		>&2 printf 'Usage: %s\n' "$usage"
+		exit 1 ;;
+
+	# Argument not prefixed with a dash
+	*) break ;;
+
+esac; shift
+done
+
+# Remaining arguments are positional parameters that are used to specify which
+# source directories to mount in the demo cluster
+declare -a pkg_dirs=()
+declare -a pkg_names=()
 for src_dir in "$@"; do
+  # shellcheck disable=SC2044
   for pkg_dir in $(find "$src_dir/src" -type f -mindepth 2 -maxdepth 2 -name '__init__.py'); do
```

Comment:
This is not related to this PR but should be easy to change:

```suggestion
  for pkg_dir in $(find "$src_dir/src" -mindepth 2 -maxdepth 2 -type f -name '__init__.py'); do
```

Adding `-type f` before `-min/maxdepth` generates the following warning:

```bash
find: warning: you have specified the global option -mindepth after the argument -type, but global options are not positional, i.e., -mindepth affects tests specified before it as well as those specified after it.  Please specify global options before other arguments.
find: warning: you have specified the global option -maxdepth after the argument -type, but global options are not positional, i.e., -maxdepth affects tests specified before it as well as those specified after it.  Please specify global options before other arguments.
```